qiskit/providers/ibmq/credentials/configrc.py

```diff
@@ -144,7 +172,9 @@ def store_credentials(
                        'Set overwrite=True to overwrite.')
         return
 
-    # Append and write the credentials to file.
+    # Clear `stored_credentials` to disallow saving multiple credentials to disk,
+    # and write the new `credentials` to file.
+    stored_credentials.clear()
```

Comment:
When saving different providers to disk, multiple ``[ibmq*]`` sections could be written, since the ``unique_id`` would be different for each one. 

To prevent that, I `clear` the `stored_credentials` before writing the new credentials, given that having multiple accounts stored on disk is discourage and causes the account methods such as ``stored_accounts``, ``load_account``, and ``delete_account`` to raise an exception. 

Should we keep this as the default behavior or revert to the previous, which would allow users to save different providers to disk (having the account methods still raise an exception of course)?

Reply:
You should not overwrite the existing content if `overwrite=True` is not specified. Also I'd prefer there remains a single `[ibmq]` section, which can be done if you pass in `default_provider` instead of saving it in `Credential`.